fiat-core/src/main/java/com/netflix/spinnaker/fiat/model/UserPermission.java

```diff
@@ -131,6 +125,18 @@ public View(UserPermission permission) {
           (Set<ServiceAccount.View>) toViews.apply(permission.getServiceAccounts());
       this.roles = (Set<Role.View>) toViews.apply(permission.getRoles());
       this.buildServices = (Set<BuildService.View>) toViews.apply(permission.getBuildServices());
+
+      this.extensionResources = new HashMap<>();
+      for (val resource : permission.extensionResources) {
+        val set =
+            this.extensionResources.computeIfAbsent(
+                resource.getResourceType(), (ignore) -> new HashSet<>());
+        val view = ((Viewable) resource).getView(permission.getRoles(), permission.isAdmin());
+        if (view instanceof Authorizable) {
```

Comment:
This is not great. The types need to be re-worked at some point. When you push a `Viewable` resource into `UserPermission`, you have no idea what kind of view it's going to give you! You have to do this check here to make sure that it was actually an `Authorizable`.

Ideally the plugin developer would know at compile-time what type their resource has to implement. Right now, they just have to magically know that their `Resource`'s `View` has to implement `Authorizable`.

Reply:
can we update View to extend/implement Authorizable in Resource or is there no common hierarchy here?

Reply:
I wish! The problem is that `ServiceAccount.View` and `Role.View` do their own thing and don't implement `Authorizable`.